jerry-core/ecma/operations/ecma-comparison.c

```diff
@@ -192,7 +192,10 @@ ecma_op_abstract_equality_compare (ecma_value_t x, /**< first operand */
       return ecma_make_boolean_value (ecma_bigint_is_equal_to_number (x, ecma_get_number_from_value (y)));
     }
 
-    return ECMA_VALUE_FALSE;
+    /* Swap values. */
+    ecma_value_t tmp = x;
+    x = y;
+    y = tmp;
```

Comment:
You can do this without allocating a third temporary variable:
```
x ^= y;
y ^= x;
x ^= y;
```

Reply:
This is copied from above. Shall I change all of them or keep this method for now?

Reply:
Rewritten.